### ISSUE TITLE
spack install does not respect the `repo` and `spec` combo inside spack.yaml

### ISSUE THREAD

**Issue by dzhang-b** [LLNL/spack#18477]
When I activate one environment, and then run `spack install`, it will only read the package file from default repo, but not the repo I defined in `spack.yaml`
 
### Steps to reproduce the issue

```console
$ spack env activate -d . # local directory has a spack.yaml file
$ spack install
...
```

```yaml
spack:
  view: false
  config:
    install_tree: /opt/vendor
  specs:
  - grpc@1.17.0  # the file is presented in both builtin and vendor repo, but builtin repo does not have 1.17 version
  repos:
    - /opt/vendor/repos/
    - $spack/var/spack/repos/builtin
```

### system info
spack version 0.15.4

**Comment by cwsmith** [comment]
From my limited experience/testing, both of these issues are resolved with v0.16.0 (and possibly in earlier versions).

**Comment by haampie** [comment]
Thanks @cwsmith, not sure why this was not closed yet
